edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -660,6 +660,16 @@ def __infer_func_call(
         _, arg_upper = arg_card
         return _bounds_to_card(CardinalityBound.ONE, max(arg_upper))
 
+    elif ir.func_shortname == sn.QualName('std', 'assert_distinct'):
```

Comment:
Why does this need special handling?

Reply:
`std::assert_distinct` preserves the cardinality of its argument, it's completely transparent.